src/Microsoft.DotNet.Darc/src/DarcLib/VirtualMonoRepo/VmrUpdater.cs

```diff
@@ -246,19 +246,28 @@ public class VmrUpdater : VmrManagerBase, IVmrUpdater
 
         reposToUpdate.Enqueue(new DependencyUpdate(mapping, targetRevision, targetVersion, null));
 
+        string commitMessage = $"Recursive update for {mapping.Name} / {GetCurrentVersion(mapping)} → {targetRevision}";
+
+        _logger.LogInformation(commitMessage);
+
+        // When we synchronize in bulk, we do it in a separate branch that we then merge into the main one
+        string syncBranchName = $"sync/{DarcLib.Commit.GetShortSha(GetCurrentVersion(mapping))}-{targetRevision}";
+        string currentBranch;
+        using (var repo = new Repository(_vmrInfo.VmrPath))
```

Comment:
Generic question - do we need to think about locking the repo somehow when manipulating it? Is there any chance of race conditions?